src/CommonUtilities/Base32.cs

```diff
@@ -37,16 +45,25 @@ public static class Base32
                     _ => throw new InvalidOperationException(), // Keeps the compiler happy.
                 };
 
-            static byte FlipByte(byte data)
-                => (byte)(
-                    (((data & 1 << 0) == 0) ? 0 : 1 << 7) |
-                    (((data & 1 << 1) == 0) ? 0 : 1 << 6) |
-                    (((data & 1 << 2) == 0) ? 0 : 1 << 5) |
-                    (((data & 1 << 3) == 0) ? 0 : 1 << 4) |
-                    (((data & 1 << 4) == 0) ? 0 : 1 << 3) |
-                    (((data & 1 << 5) == 0) ? 0 : 1 << 2) |
-                    (((data & 1 << 6) == 0) ? 0 : 1 << 1) |
-                    (((data & 1 << 7) == 0) ? 0 : 1 << 0));
-        }
+        /// <summary>
+        /// Converts each group (fixed number) of items into a new item
+        /// </summary>
+        /// <typeparam name="TSource">Type of source items</typeparam>
+        /// <typeparam name="TGroupItem">Intermediate type</typeparam>
+        /// <typeparam name="TResult">Type of the resultant items</typeparam>
+        /// <param name="ts">The source enumerable of type <typeparamref name="TSource"/>.</param>
+        /// <param name="itemsPerGroup">The size of each group to create out of the entire enumeration. The last group may be smaller.</param>
+        /// <param name="groupMemberFunc">The function that prepares each <typeparamref name="TSource"/> into the value expected by <paramref name="groupResultFunc"/>. Its parameters are <typeparamref name="TSource"/> and the index if that item (starting from zero) within each grouping.</param>
+        /// <param name="groupResultFunc">The function that creates the <typeparamref name="TResult"/> from each group of <typeparamref name="TGroupItem"/>.</param>
+        /// <returns>An enumeration of <typeparamref name="TResult"/> from all of the groups.</returns>
+        static IEnumerable<TResult> ConvertGroup<TSource, TGroupItem, TResult>(
```

Comment:
Maybe this should be named `StreamedTakeAndGroup`? I'm notorious for bad names, btw